File: python/2348.py
# Link: https://leetcode.com/problems/number-of-zero-filled-subarrays/description/
# Runtime: 24 ms
# Memory: 28.26 MB


class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        number_of_subarrays, streak = 0, 0
        for val in nums:
            if not val:
                streak += 1
                number_of_subarrays += streak
            else:
                streak = 0

        return number_of_subarrays

File: c++/2348.cpp
// Link: https://leetcode.com/problems/number-of-zero-filled-subarrays/description/
// Runtime: 0 ms
// Memory: 111.34 MB

class Solution {
public:
  long long zeroFilledSubarray(const std::vector<int> &nums) {
    long long numberOfSubarrays = 0;
    int streak = 0;
    for (const auto &val : nums) {
      if (val == 0) {
        streak++;
        numberOfSubarrays += streak;
      } else {
        streak = 0;
      }
    }

    return numberOfSubarrays;
  }
};